spec/services/duplicate_application_spec.rb

```diff
@@ -22,94 +22,415 @@
+          )
         end
+      end
+    end
+
+    context 'when field is HESA ethnicities' do
+      context 'when ethnic background is nil' do
+        it 'carries over as the same ethnic data for' do
+          @original_application_form.update!(
+            recruitment_cycle_year: 2023,
+            equality_and_diversity: @original_application_form.equality_and_diversity.merge(
+              hesa_ethnicity: nil,
+              ethnic_background: nil,
+            ),
+          )
 
-        it 'marks reference as incomplete' do
-          expect(duplicate_application_form).not_to be_references_completed
+          expect(duplicate_application_form.equality_and_diversity['hesa_ethnicity']).to be_nil
+          expect(duplicate_application_form.equality_and_diversity['ethnic_background']).to be_nil
         end
       end
 
-      context 'some references are cancelled but candidate still has two valid' do
-        before do
-          @original_application_form.application_references.last.cancelled!
+      context 'when 2022 cycle application' do
+        [
+          { hesa_code: '10', expected_conversion: '160', ethnic_background: 'White' },
+          { hesa_code: '15', expected_conversion: '163', ethnic_background: 'Gypsy or Traveller' },
+          { hesa_code: '21', expected_conversion: '121', ethnic_background: 'Black or Black British - Caribbean' },
+          { hesa_code: '22', expected_conversion: '120', ethnic_background: 'Black or Black British - African' },
+          { hesa_code: '29', expected_conversion: '139', ethnic_background: 'Other Black background' },
+          { hesa_code: '31', expected_conversion: '103', ethnic_background: 'Asian or Asian British - Indian' },
+          { hesa_code: '32', expected_conversion: '104', ethnic_background: 'Asian or Asian British - Pakistani' },
+          { hesa_code: '33', expected_conversion: '100', ethnic_background: 'Asian or Asian British - Bangladeshi' },
+          { hesa_code: '34', expected_conversion: '101', ethnic_background: 'Chinese' },
+          { hesa_code: '39', expected_conversion: '119', ethnic_background: 'Other Asian background' },
+          { hesa_code: '41', expected_conversion: '142', ethnic_background: 'Mixed - White and Black Caribbean' },
+          { hesa_code: '42', expected_conversion: '141', ethnic_background: 'Mixed - White and Black African' },
+          { hesa_code: '43', expected_conversion: '140', ethnic_background: 'Mixed - White and Asian' },
+          { hesa_code: '49', expected_conversion: '159', ethnic_background: 'Other Mixed background' },
+          { hesa_code: '50', expected_conversion: '180', ethnic_background: 'Arab' },
+          { hesa_code: '80', expected_conversion: '899', ethnic_background: 'Other Ethnic background' },
+          { hesa_code: '90', expected_conversion: '997', ethnic_background: 'Not known' },
+          { hesa_code: '98', expected_conversion: '998', ethnic_background: 'Prefer not to say' },
```

Comment:
I moved all this code to the HESA converter. I think I will just test here some edge cases but testing just that the HesaConverter is being called. I didn't made the changes yet. I will do tomorrow.